tests/foreman/cli/test_docker.py

```diff
@@ -1452,84 +1482,87 @@ def test_positive_delete_by_id(self):
 
         :CaseLevel: System
         """
-        for url in (settings.docker.external_url,
-                    settings.docker.get_unix_socket_url()):
-            with self.subTest(url):
-                compute_resource = make_compute_resource({
-                    'provider': DOCKER_PROVIDER,
-                    'url': url,
-                })
-                self.assertEqual(compute_resource['url'], url)
-                self.assertEqual(compute_resource['provider'], DOCKER_PROVIDER)
-                ComputeResource.delete({'id': compute_resource['id']})
-                with self.assertRaises(CLIReturnCodeError):
-                    ComputeResource.info({'id': compute_resource['id']})
+        docker_image = settings.docker.docker_image
+        with VirtualMachine(source_image=docker_image) as docker_host:
+            docker_host.create()
+            url = 'http://{0}:2375'.format(docker_host.ip_addr)
+            compute_resource = make_compute_resource({
+                'provider': DOCKER_PROVIDER,
+                'url': url,
+            })
+            self.assertEqual(compute_resource['url'], url)
+            self.assertEqual(compute_resource['provider'], DOCKER_PROVIDER)
+            ComputeResource.delete({'id': compute_resource['id']})
```

Comment:
hey..this is CLI instance, so  ComputeResource at that point is a simple dictionary